shared/src/commonMain/kotlin/org/hyperskill/app/track_list/presentation/TrackListActionDispatcher.kt

```diff
@@ -0,0 +1,103 @@
+package org.hyperskill.app.track_list.presentation
+
+import org.hyperskill.app.analytic.domain.interactor.AnalyticInteractor
+import org.hyperskill.app.core.presentation.ActionDispatcherOptions
+import org.hyperskill.app.freemium.domain.interactor.FreemiumInteractor
+import org.hyperskill.app.profile.domain.interactor.ProfileInteractor
+import org.hyperskill.app.progresses.domain.interactor.ProgressesInteractor
+import org.hyperskill.app.sentry.domain.interactor.SentryInteractor
+import org.hyperskill.app.sentry.domain.model.transaction.HyperskillSentryTransactionBuilder
+import org.hyperskill.app.study_plan.domain.repository.CurrentStudyPlanStateRepository
+import org.hyperskill.app.track.domain.interactor.TrackInteractor
+import org.hyperskill.app.track.domain.model.TrackWithProgress
+import org.hyperskill.app.track_list.presentation.TrackListFeature.Action
+import org.hyperskill.app.track_list.presentation.TrackListFeature.Message
+import ru.nobird.app.presentation.redux.dispatcher.CoroutineActionDispatcher
+
+class TrackListActionDispatcher(
+    config: ActionDispatcherOptions,
+    private val analyticInteractor: AnalyticInteractor,
+    private val sentryInteractor: SentryInteractor,
+    private val trackInteractor: TrackInteractor,
+    private val progressesInteractor: ProgressesInteractor,
+    private val profileInteractor: ProfileInteractor,
+    private val freemiumInteractor: FreemiumInteractor,
+    private val currentStudyPlanStateRepository: CurrentStudyPlanStateRepository
+) : CoroutineActionDispatcher<Action, Message>(config.createConfig()) {
+    override suspend fun doSuspendableAction(action: Action) {
+        when (action) {
+            is Action.Initialize -> {
+                val sentryTransaction =
+                    HyperskillSentryTransactionBuilder.buildPlaceholderNewUserScreenRemoteDataLoading()
+                sentryInteractor.startTransaction(sentryTransaction)
+
+                val tracks = trackInteractor
+                    .getAllTracks()
+                    .getOrElse {
+                        sentryInteractor.finishTransaction(sentryTransaction, it)
+                        return onNewMessage(Message.TracksLoaded.Error)
+                    }
+                    .filter { it.isBeta.not() }
```

Comment:
I'll ask

Reply:
Natasha answered that it's ok to show beta tracks. We need to load all tracks anyway, because backend doesn't have sorting parameter for `is_beta` field. Therefore I removed this filtering